File: smo-inspection_backend/src/main/java/com/swire/smo/inspection/domain/QuestionDetails.java
package com.swire.smo.inspection.domain;

import java.util.List;

import lombok.Data;

/**
 *
 * @author Krishna
 */
@Data
public class QuestionDetails {
	
    private Integer id;
	
	private Integer QuestionId;

	private String reply;

	private String unsafeAactAction;

	private String actionRequired;

	private String personInCharge;

	private String targetCompletionDate;

	private String statusOfAction;

	private String dateOfActionCompleted;

	private boolean IsIssueResolved;
	
	private Integer mainHeadingId;
	
	private Integer subHeadingId;
	
	private List<AnswerDetails> answerDetaiils;

}

File: smo-inspection_frontend/src/app/QuestionDetails.ts
export class QuestionDetails{
    public actionRequired: string;
    public dateOfActionCompleted: string;
    public isIssueResolved: boolean;
    public mainHeadingId: number;
    public personInCharge: string;
    public questionId: number;
    public reply: string;
    public statusOfAction:string;
    public subHeadingId:number;
    public targetCompletionDate:string;
    public unsafeAactAction:string;
constructor(
    actionRequired: string,
    dateOfActionCompleted: string,
    isIssueResolved: boolean,
    mainHeadingId: number,
    personInCharge: string,
    questionId: number,
    reply: string,
    statusOfAction:string,
    subHeadingId:number,
    targetCompletionDate:string,
    unsafeAactAction:string
    ) {
        this.actionRequired=actionRequired;
        this.dateOfActionCompleted=dateOfActionCompleted;
        this.isIssueResolved=isIssueResolved;
        this.mainHeadingId=mainHeadingId;
        this.personInCharge=personInCharge;
        this.questionId=questionId;
        this.reply=reply;
        this.statusOfAction=statusOfAction;
        this.subHeadingId=subHeadingId;
        this.targetCompletionDate=targetCompletionDate;
        this.unsafeAactAction=unsafeAactAction;
    }
}